src/handler/error_handler.cpp

```diff
@@ -9,8 +9,9 @@ size_t ErrorHandler::read_data(char* buf, size_t n)
 {
     std::string res = res_.to_string();
     LOG(DEBUG) << res;
-    std::memcpy(buf, &res, n);
-    return (res.size());
+    size_t to_copy = std::min(res.size(), n);
+    std::memcpy(buf, res.c_str(), to_copy);
```

Comment:
```suggestion
    std::memcpy(buf, res.data(), to_copy);
```

Only because of binary data we could have issues using null terminated strings here